README.md

```diff
@@ -45,20 +56,69 @@ You can either keep running the code from source, using `cargo run --release --b
 
 The logging of mintlayer-core is configured via the `RUST_LOG` environment variable. All log messages are printed to the terminal screen; we prefer simplicity over complicated log machinery. For example, to see all logs of the `info` level and above (the default level for normal operation), you can run the node with `RUST_LOG=info cargo run --bin node-daemon -- testnet`. If you're facing an issue, it's recommended to use `RUST_LOG=debug` instead. We recommend using these commands that not only print the logs on the screen, but also write them to a file in case you face an issue. On Linux, this can be achieved using `tee` as shown below.
 
+If the `RUST_LOG` environment variable is not specified, the log level `info` will be used by default.
+
 Here are the commands as recommended for different scenarios:
 
+#### Assuming you're using the source code
+
+Every release has a tag and a release branch. Make sure you checkout the release you need. For example, if you need v0.3.0, you should first run the following to check out the release branch:
+
+```sh
+git checkout release-v0.3.0
+```
+
+or to checkout the tag:
+
+```sh
+git checkout tags/v0.3.0
+```
+
+Release branches are more recommended than tags, because they get necessary security patches, if any. Tags are just markers.
+
+After having checked out the version you need, the following is how to run the software.
+
 For normal operation (replace testnet by mainnet to run the mainnet node)
+You don't need to specify RUST_LOG environment variable for logging, since it's by default set to be `INFO`.
```

Comment:
The line `For normal operation (replace testnet by mainnet to run the mainnet node)` looks off, perhaps it should be made a header

Also a nitpick: IMO `replace with` would sound better than `replace by` in this context.